### PR TITLE
refactor: Restructure log manager responsibility.

### PR DESCRIPTION
**Pull request by Isakdl**
# Changes

Refactor the logging decisions to be made inside the log manager instead of outside.

Related: https://github.com/serverpod/serverpod/issues/2450

## Pre-launch Checklist

- [ ] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [ ] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [ ] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [ ] All existing and new tests are passing.
- [ ] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

Breaking internally annotated methods.

### CODE REVIEW

File: packages/serverpod/lib/src/server/log_manager/log_manager.dart [hunk 1]
```diff
@@ -123,10 +122,44 @@ class LogManager {
 
   /// Logs a query, depending on the session type it will be logged directly
   /// to the database or stored in the temporary cache until the session is
-  /// closed. Call [shouldLogQuery] to check if the entry should be logged
+  /// closed. Call [_shouldLogQuery] to check if the entry should be logged
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/log_manager/log_manager.dart:154]: Suggestion: Remove instruction that `_shouldLogQuery` should be called before this method is called.

File: packages/serverpod/lib/src/server/log_manager/log_manager.dart [hunk 2]
```diff
@@ -123,10 +122,44 @@ class LogManager {
...
-  Future<void> logQuery(Session session, QueryLogEntry entry) async {
+  Future<void> logQuery(
+    Session session, {
+    required String query,
+    required Duration duration,
+    required int? numRowsAffected,
+    required String? error,
+    required StackTrace stackTrace,
+  }) async {
+    var executionTime = duration.inMicroseconds / (1000 * 1000.0);
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/log_manager/log_manager.dart:165]: Suggestion: Add a constant for the multiplication done here.

We don't need to do this multiplication over and over again at runtime.

File: packages/serverpod/lib/src/server/serverpod.dart [hunk 3]
```diff
@@ -22,6 +22,9 @@ import '../cache/caches.dart';
 import '../generated/endpoints.dart' as internal;
 import '../generated/protocol.dart' as internal;
 
+/// The default server id if none is specified on startup.
+const defaultServerId = 'default';
+
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/serverpod.dart:27]: Question: When reviewing commit by commit and looking at the second commit this doesn't seem to be used. Can it be removed?

File: packages/serverpod/lib/src/server/log_manager/log_manager.dart [hunk 4]
```diff
@@ -115,7 +113,36 @@ class LogManager {
   /// closed. Call [shouldLogEntry] to check if the entry should be logged
   /// before calling this method. This method can be called asynchronously.
   @internal
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/log_manager/log_manager.dart:111]: Suggestion: Remove `shouldLogEntry` from documentation.

File: packages/serverpod/lib/src/server/log_manager/log_manager.dart [hunk 5]
```diff
@@ -209,7 +203,42 @@ class LogManager {
...
+  Future<void> logMessage(
+    Session session, {
+    required String endpointName,
+    required String messageName,
+    required int messageId,
+    required Duration duration,
+    required String? error,
+    required StackTrace? stackTrace,
+  }) async {
+    var executionTime = duration.inMicroseconds / (1000 * 1000.0);
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/log_manager/log_manager.dart:215]: Suggestion: again, please add a constant for the multiplication here.

File: packages/serverpod/lib/src/server/log_manager/log_manager.dart [hunk 6]
```diff
@@ -209,7 +203,42 @@ class LogManager {
...
+
+    var slow = executionTime >=
+        settings.getLogSettingsForSession(session).slowSessionDuration;
+
+    var shouldLog = _shouldLogMessage(
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/log_manager/log_manager.dart:220]: Question: are the shouldLog conditions the same for the different logging reasons?

**Review by SandPod — CHANGES_REQUESTED** [review]
Nice cleanup! Good to have that logic in the same place.

A couple of suggestions and maybe a minor fix.

File: packages/serverpod/lib/src/server/log_manager/log_manager.dart (hunk 6 above)
Comment (Isakdl) [reply packages/serverpod/lib/src/server/log_manager/log_manager.dart:220]: No!

**Review by SandPod — APPROVED** [review]
LGTM! 🎉 🚀

**Pull request merged**